### PR TITLE
Enable back button if the pin is prompted but not during creation

### PR DESCRIPTION
**Pull request by osdnk**
Fixes RNBW-4090
Figma link (if any):
trivial fix
## Screen recordings / screenshots

https://user-images.githubusercontent.com/25709300/183044343-349457c3-c56d-493f-a515-64a9535651c4.mov

### CODE REVIEW

**Comment by tchayen** [comment]
Looks like back button worked for me on the pin creation and it shouldn't?

https://user-images.githubusercontent.com/5769281/183105589-e1e1cee4-6929-4b44-9af5-7afd50106346.mp4

**Review by tchayen — APPROVED** [review]
Tested and works for creating a wallet.

File: src/screens/PinAuthenticationScreen.js [hunk 1]
```diff
@@ -31,8 +31,8 @@ const MAX_ATTEMPTS = 10;
 const TIMELOCK_INTERVAL_MINUTES = 5;
 
 const PinAuthenticationScreen = () => {
-  useBlockBackButton();
   const { params } = useRoute();
+  useBlockBackButton(!params.validPin);
```
[the remark this answers is not in the document]
Comment (osdnk) [reply src/screens/PinAuthenticationScreen.js:35]: you cannot hook conditionally. Do you have beet ideas?

File: src/hooks/useBlockBackButton.ts [hunk 2]
```diff
@@ -3,8 +3,11 @@ import { NativeModules } from 'react-native';
 
 const { RNBackHandler } = NativeModules;
 
-export function useBlockBackButton() {
+export function useBlockBackButton(block: boolean) {
   useEffect(() => {
+    if (!block) {
```
Comment (terrysahaidak) [inline src/hooks/useBlockBackButton.ts:8]:
```suggestion
export function useBlockBackButton(shouldBlock: boolean) {
  useEffect(() => {
    if (!shouldBlock) {
```

**Review by terrysahaidak — APPROVED** [review]
LGTM

**Pull request merged**